apps/idos-enclave/src/lib/enclave.js

```diff
@@ -254,6 +254,69 @@ export class Enclave {
       .map((credential) => credential.id);
   }
 
+  async filterCredentials(credentials, privateFieldFilters) {
+    const filterCredentials = async (credentials, privateFieldFilters) => {
+      function findValueByKey(obj, key) {
+        if (obj[key]) {
+          return obj[key];
+        }
+
+        for (const prop in obj) {
+          if (obj[prop] && typeof obj[prop] === "object") {
+            const result = findValueByKey(obj[prop], key);
+
+            if (result !== undefined) {
+              return result;
+            }
+          }
+        }
+
+        return undefined;
+      }
+
+      const filterByKeys = (content, keys, filterCriteria, include = true) => {
+        for (const key of keys) {
+          const matchedValue = findValueByKey(content, key);
+          const isMatch = filterCriteria[key].includes(matchedValue);
+          if (include && !isMatch) {
+            return false;
+          }
+          if (!include && isMatch) {
+            return false;
+          }
+        }
+        return true;
```

Comment:
This feels way too complicated.

Can we use some helper methods (from lodash or some other TS helper lib)?